pandas/core/base.py

```diff
@@ -902,7 +902,12 @@ def __iter__(self):
         (for str, int, float) or a pandas scalar
         (for Timestamp/Timedelta/Interval/Period)
         """
-        return iter(self.tolist())
+        if is_datetimelike(self._values):
```

Comment:
can you add a comment here on the impl, noting that we are explicity making element iterators

Reply:
Added. If I understood you correctly what you are asking for.